src/compile/mark/encode/tooltip.ts

```diff
@@ -121,7 +121,7 @@ export function tooltipData(
       }).signal;
     }
 
-    value ??= textRef(fieldDef, config, expr).signal;
+    value ??= textRef(fieldDef, {...config, ...config?.tooltipFormat}, expr).signal;
```

Comment:
nit: Let's add `config = {...config, ...config?.tooltipFormat}` to the top of `function tooltipData` so we don't make the copy for every `add` call before merging.